### ISSUE TITLE
Please help or fault-find issue with USB hardware link in Linux?

### ISSUE THREAD

**Issue by stutzjr** [VCVRack/Rack#844]
I typically use VCV in Windows 10 on my desktop PC and am able to interface with my physical modular system with a Roland Bitrazer (Aira Custom Modular).  I have attached an image showing a worked example where I send an audio signal out to the Bitrazer/Aira and modulate it with a physical envelope+LPG and then send it back to VCV Rack via one of the 6 "Griffer" inputs.  Barring CPU limitations, this seems to work fine in Windows using "DirectSound".

![Image of VCV example](https://i.imgur.com/Yrlpuum.jpg)

When I try to do the same thing using my laptop running Fedora Linux, VCVrack only seems to have access to the "ALSA" audio driver, and things don't go so well.  When I attempt to select the audio device "AIRA Modular: 10 in/10 out"* after I click on it the "audio device" remains blank and nothing happens.  (* I don't remember the exact text here, but there is only one listing, whereas in Windows VCV under DirectSound there is a separate listing for each pair of inputs / outputs.

I really don't have a lot of knowledge / motivation when it comes to tacking Linux compatibility issues, so I guess I will just ask if anyone else can report success or similar issues with Roland Aira or Expert Sleepers hardware in Linux?

**Comment by danballance** [comment]
I think you may possibly need to compile VCV with Jack support and use Jack to route your audio. That's what I do in order to run VCV alongside my Bigwig DAW and to route modulation signals out to my hardware modular via ADAT and the Expert Sleepers ES-3.

**Comment by stutzjr** [comment]
Cheers danballance, I figured this might be the case.  I'm still reorganising my workroom so I won't be using the laptop+modular combination for a while yet anyway.  I'll wait and see how things change with v 0.6 before getting myself lost in the dependency purgatory that I seem to rediscover whenever I try to compile Linux projects from source...

**Comment by danballance** [comment]
I do know what you mean - it's  a pain having to compile to get Jack support. For what it's worth I found the process pretty easy when I last ran it, but it does of course depend on your level of familiarity with Linux. This is one of those example where I worry we are going to lose Linux users because the technical skills to run VCV on Linux are too high (for something creative and fun like making music).

IIRC there are technical issues though with regard to making a generic binary that works with rack on all systems. It's a lot easier to just build the support locally on your own machine. It's a shame though because my guess is this is going to reduce Linux adoption.

**Comment by stutzjr** [comment]
As unfortunate as it is, I think generally the boat for Linux adoption probably sailed some time ago.

In terms of VCV however, I was able to simply download and unzip the 0.5.1 from the website and I was good to go with my Fedora laptop - pretty straightforward and easy really.  I only came across an issue when I wanted to try the external hardware interface.  This may not have been an issue at all if Roland had opted to support Linux to begin with - I know that's not going to happen though, as the Aira custom modular units were discontinued.
